store/index-refactor.js

```diff
@@ -27,6 +33,13 @@ export const getters = {
             return "";
         }
     },
+
+    getColumnNames(p_state) {
+
+        // Returns list of columns from the loaded data table
+        return ( 0 === p_state.dataTable.length ) ? [] : Object.keys(p_state.dataTable[0]);
```

Comment:
Is there any valid path for this getter to be called before a `dataTable` has been loaded?

Reply:
This is not necessarily a straightforward question to answer at the moment because our refactor of the pages for the annotation tool is still in progress. The most direct answer is that it depends on how a getter such as this is used and where – and how gracefully we want to handle failures if the components we hand invalid values to do not.

If a user loads something like the categorization page directly – from a previous browser session for instance and Vue server is still up and running but with a wiped Vuex store – a check like this offers a more graceful failure than what would occur if an empty `dataTable`'s 0th index is accessed.

With the conditional check, the empty array would be offered to the `column-linking-table` component's Bootstrap table. Otherwise the table would be handed `undefined` (or the page itself would crash).

Reply:
Ah ok, that's a really good point. So there is at least one way we know how this getter could be called before a dataTable has been loaded (the one you described). In that case my follow-up question is: shouldn't we then also test in the components that they are in fact OK with receiving these empty arrays? Not for this PR. But if we say: you can reload the page and we'll make sure that the page doesn't crash and instead give you a valid but empty UI, then we should test that this is in fact true. 

But for now, agreed that the check makes sense here.

Reply:
>shouldn't we then also test in the components that they are in fact OK with receiving these empty arrays? Not for this PR. But if we say: you can reload the page and we'll make sure that the page doesn't crash and instead give you a valid but empty UI, then we should test that this is in fact true.

Absolutely true. Relevant component tests and should test for invalid input if such a scenario exists. And I would say this would extend to page tests, so we can give our users some helpful feedback as to why something has gone wrong and how they might fix it.